packages/sdks/web-js-sdk/src/enhancers/withFlowNonce/index.ts

```diff
@@ -0,0 +1,71 @@
+import { CreateWebSdk } from '../../sdk';
+import { AfterRequestHook, BeforeRequestHook } from '../../types';
+import { addHooks } from '../helpers';
+import {
+  FLOW_NEXT_PATH,
+  FLOW_NONCE_HEADER,
+  FLOW_NONCE_PREFIX,
+  FLOW_START_PATH,
+} from './constants';
+import {
+  cleanupExpiredNonces,
+  extractFlowNonce,
+  getExecutionIdFromRequest,
+  getFlowNonce,
+  setFlowNonce,
+} from './helpers';
+import { FlowNonceOptions } from './types';
+
+/**
+ * Adds flow nonce handling to the SDK
+ */
+export const withFlowNonce =
+  <T extends CreateWebSdk>(createSdk: T) =>
+  (config: Parameters<T>[0] & FlowNonceOptions): ReturnType<T> => {
+    const {
+      enableFlowNonce = true,
+      nonceStoragePrefix = FLOW_NONCE_PREFIX,
+      ...sdkConfig
+    } = config;
+
+    if (!enableFlowNonce) {
+      return createSdk(sdkConfig) as ReturnType<T>;
+    }
+
+    cleanupExpiredNonces(nonceStoragePrefix);
```

Comment:
Why do we want to clean all expired keys on SDK init and not do it lazily?

Reply:
We do it lazy, thought to add here for other execution ids